src/main/java/org/spongepowered/api/command/spec/CommandSpec.java

```diff
@@ -254,7 +291,7 @@ public CommandSpec build() {
             }
 
             return new CommandSpec(this.args, this.executor, this.description, this.extendedDescription, this.permission,
```

Comment:
Well, the whole point of an API is that it should pretty much hold the developer's hand! It's supposed to provide type safety!

Would it be better if you provided two executor methods instead and make use of compile time safety? For the current case, no source type filtering (though, this will cause a warning in current code, but that's what bleeding is for, right?)

```
public Builder executor(CommandExecutor<CommandSource> executor) 
```

and, for what you're trying to do:

```
public <T extends CommandSource> Builder executor(CommandExecutor<? super T> executor, Class<T> sourceType) 
```

This will then make the compiler check that the source type is correct. Having a two argument method does feel like it goes against the grain of the builder pattern, but given the source type and the fact Java has type erasure, maybe it's a better solution?